packages/message/src/DeadLetterMessage.ts

```diff
@@ -1,21 +1,47 @@
-import { SQSRecord, EventBridgeEvent } from 'aws-lambda';
+//@ts-check
+import { SQSRecord } from 'aws-lambda';
 
 import { parseSQSMessageBody } from '@cumulus/aws-client/SQS';
 import { CumulusMessage } from '@cumulus/types/message';
+import { isEventBridgeEvent, StepFunctionEventBridgeEvent } from '@cumulus/aws-client/Lambda';
 import Logger from '@cumulus/logger';
 
 import { getCumulusMessageFromExecutionEvent } from './StepFunctions';
 
 const log = new Logger({ sender: '@cumulus/DeadLetterMessage' });
+interface DLQRecord extends SQSRecord {
+  error?: string
+}
+export type { DLQRecord };
 
-type StepFunctionEventBridgeEvent = EventBridgeEvent<'Step Functions Execution Status Change', { [key: string]: string }>;
-type UnwrapDeadLetterCumulusMessageReturnType = (
+type UnwrapDeadLetterCumulusMessageInputType = (
   StepFunctionEventBridgeEvent
-  | AWS.SQS.Message
-  | SQSRecord
+  | DLQRecord | SQSRecord
   | CumulusMessage
 );
 
+/**
+ * Bare check for CumulusMessage Shape
+ *
+ * @param {{ [key: string]: any }} message
+ * @returns {message is CumulusMessage}
+ */
+const isCumulusMessageLike = (message: Object): message is CumulusMessage => (
+  message instanceof Object
+  && 'cumulus_meta' in message
+);
+
+/**
+ * Bare check for SQS message Shape
+ *
+ * @param {{ [key: string]: any }} message
+ * @returns {message is DLQRecord}
+ */
+export const isSQSRecordLike = (message: Object): message is DLQRecord | SQSRecord => (
```

Comment:
Thinking about isEventBridgeEvent/isSQSRecordLike, and typeguards in general: 

Should this be in the aws-client package?   SQSRecord doesn't seem to map to `@cumulus/message` as a package?

Reply:
resolved